README.md

```diff
@@ -48,35 +48,36 @@ authorization is destroyed.
 
 A provider in Torii is anything a user can authenticate against. This could be an
 OAuth 2.0 endpoint, your own login mechanism, or an SDK like Facebook Connect.
-Authenticating against a **provider** is done via the `torii` property, which is injected
-into routes:
+Authenticating against a **provider** is done via the `torii` property, which must be
+explicitly injected into routes:
 
 ```hbs
 {{! app/templates/post.hbs }}
-{{#if hasFacebook}}
+{{#if this.hasFacebook}}
   {{partial "comment-form"}}
 {{else}}
-  <a href="#" {{action 'signInToComment'}}>
+  <a href="#" {{on "click" this.signInToComment}}>
     Sign in to comment
   </a>
 {{/if}}
 ```
 
 ```JavaScript
 // app/routes/post.js
-export default Ember.Route.extend({
-  actions: {
-    signInToComment: function(){
-      var controller = this.controllerFor('post');
-      // The provider name is passed to `open`
-      this.get('torii').open('facebook-connect').then(function(authorization){
-        // FB.api is now available. authorization contains the UID and
-        // accessToken.
-        controller.set('hasFacebook', true);
-      });
-    }
+export default class PostRoute extends Route {
+  @service torii;
+
+  @action
+  signInToComment() {
+    var controller = this.controllerFor('post');
+    // The provider name is passed to `open`
+    this.torii.open('facebook-connect').then(function(authorization){
```

Comment:
Not sure if we also want to change examples using promises to use `async` / `await` syntax or if  these should be left as is.

Reply:
Discussion starting from [here ](https://discord.com/channels/480462759797063690/483601670685720591/1014562222829215754)in the ember discord is leading me to think that leaving these as-is is the move.